### PR TITLE
Update Mongo driver to `v6.9.0`

### PR DESCRIPTION
**Pull request by nick-funk**
## What does this PR do?

- Upgrades Mongo to the latest driver (v6.9.0)

## These changes will impact:

- [ ] commenters
- [ ] moderators
- [ ] admins
- [X] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

N/A

## How do I test this PR?

This touches all of Coral's data retrieval and persistence, all features must be re-tested in a regression test.

## Were any tests migrated to React Testing Library?

No

## How do we deploy this PR?

- Deploy to staging/testing environments
- Perform a full regression test
    - Fix any critical bugs that are found
- Issue out a release

### CODE REVIEW

File: server/src/core/server/models/comment/metrics.ts [hunk 1]
```diff
@@ -116,10 +118,12 @@ export async function retrieveTodayCommentMetrics(
       continue;
     }
 
+    const mod = moderator;
```
Comment (kabeaty) [inline server/src/core/server/models/comment/metrics.ts:121]: just curious why we're assigning this here now? Looks like we're still using `moderator` on line 124 though? (Same question about similar change below in the file too)

File: server/src/core/server/models/helpers/query.ts [hunk 2]
```diff
@@ -65,15 +70,16 @@ export default class Query<T> {
    * orderBy will apply sorting to the query filter when executed.
    * @param sort the sorting option for the documents
    */
-  public orderBy(sort: object): Query<T> {
+  public orderBy(sort: { [key: string]: SortDirection }): Query<T> {
+    // todo: merge sort's together
```
Comment (kabeaty) [inline server/src/core/server/models/helpers/query.ts]: Is this still a TODO?

File: server/src/core/server/services/mongodb/index.ts [hunk 3]
```diff
@@ -6,7 +6,10 @@ import logger from "coral-server/logger";
 async function createMongoClient(mongoURI: string): Promise<MongoClient> {
   try {
     return await MongoClient.connect(mongoURI, {
-      useNewUrlParser: true,
+      // believe we don't need this since the new driver only uses
+      // the new URL parser. But am leaving this here in case we have
+      // issues with URL's and want to investigate into it.
+      // useNewUrlParser: true,
```
Comment (kabeaty) [inline server/src/core/server/services/mongodb/index.ts:12]: 👍

File: server/src/core/server/models/site/index.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { identity, isNumber } from "lodash";
-import { MongoError } from "mongodb";
+import { Collection, MongoError } from "mongodb";
 import { v4 as uuid } from "uuid";
```
Comment (kabeaty) [inline server/src/core/server/models/site/index.ts:4]: I'm seeing a warning in the console: "(node:85676) [MONGODB DRIVER] Warning: cursor.count is deprecated and will be removed in the next major version, please use `collection.estimatedDocumentCount` or `collection.countDocuments` instead". 

Looks like we are using deprecated `count()` on line 135 in this file. Should we update to one of the suggested replacements now?

**Review by kabeaty — APPROVED** [review]
Everything seems to be working great with the update! I left a few questions throughout.

File: server/src/core/server/models/comment/metrics.ts (hunk 1 above)
Comment (nick-funk) [reply server/src/core/server/models/comment/metrics.ts:121]: it's a bug in typescript, it sometimes fails to process types after a `continue` op. If you re-assign, it works. So I did that for the few lines that were "erroring" (but compile and run fine).

File: server/src/core/server/models/helpers/query.ts (hunk 2 above)
Comment (nick-funk) [reply server/src/core/server/models/helpers/query.ts]: nope! I "sorted" this earlier! I will update!

Comment (nick-funk) [reply server/src/core/server/models/helpers/query.ts]: fixed!

File: server/src/core/server/models/site/index.ts (hunk 4 above)
Comment (nick-funk) [reply server/src/core/server/models/site/index.ts:4]: looking into this, will see if I can update the query

Comment (nick-funk) [reply server/src/core/server/models/site/index.ts:4]: yup, was an easy fix, done!

**Comment by rigperro** [comment]
Very nice! What mongo version are you looking to use with this?

**Pull request merged**